### PR TITLE
Biosys 92 add ordering search record end point

### PR DESCRIPTION
**Pull request by serge-gaia**
See https://decbugs.com/view.php?id=6866
* just added the json search/ordering implemented in the `dataset/{id}/records` end-point to the `record/` endpoint 
* some tests

### CODE REVIEW

File: biosys/apps/main/api/views.py [hunk 1]
```diff
@@ -350,6 +349,26 @@ def get_dataset(self, request, *args, **kwargs):
...
+                    'data': self.dataset.schema.field_names,
+                    'source_info': ['file_name', 'row']
+                }
+
+                queryset = search_json_fields(queryset, field_info, search_param)
+
+            ordering_param = self.request.query_params.get('ordering')
+            if ordering_param is not None:
+                queryset = order_by_json_field(queryset, 'data', self.dataset.schema.field_names, ordering_param)
+                queryset = order_by_json_field(queryset, 'source_info', ['file_name', 'row'], ordering_param)
```
Comment (tony-gaia) [inline biosys/apps/main/api/views.py:368]: perhaps do a check that the ordering_param is 'file_name" or 'row' before calling this function.

File: biosys/apps/main/tests/api/test_generic_record.py [hunk 2]
```diff
@@ -669,3 +670,161 @@ def test_excel_type(self):
...
+            ['What', 'When', 'Who'],
+            ['Crashed the db', '2018-02-14', 'Serge'],
+            ['Restored the db', '2018-02-14', 'Shay']
+        ])
+
+        dataset2 = self._create_dataset_and_records_from_rows([
+            ['What', 'When', 'Latitude', 'Longitude'],
+            ['Canis lupus', '2018-02-14', -32.0, 115.75],
+            ['Chubby bat', '2017-05-18', -34.4, 116.78],
+            ['Chubby Serge', '2017-05-18', -34.4, 116.78]
```
Comment (tony-gaia) [inline biosys/apps/main/tests/api/test_generic_record.py:735]: LOL!

File: biosys/apps/main/tests/api/test_generic_record.py [hunk 3]
```diff
@@ -669,3 +670,161 @@ def test_excel_type(self):
...
+        ])
+
+        client = self.custodian_1_client
+        url = reverse('api:record-list')
+
+        # search Serge in dataset1
+        resp = client.get(url, {'search': 'Serge', "dataset__id": dataset1.pk})
```
Comment (tony-gaia) [inline biosys/apps/main/tests/api/test_generic_record.py]: just use single quotes

File: biosys/apps/main/api/views.py (hunk 1 above)
Comment (serge-gaia) [reply biosys/apps/main/api/views.py:368]: I think that the ordering parameter should be of form: `source_info__row` or `data__col`

**Pull request merged**